xbmc/music/tags/MusicInfoTag.cpp

```diff
@@ -830,6 +848,14 @@ void CMusicInfoTag::Archive(CArchive& ar)
     ar << m_lastPlayed;
     ar << m_dateAdded;
     ar << m_strComment;
+    ar << (int)m_musicRoles.size();   
```

Comment:
this can result in data-loss, .size() returns uint64_t on some platforms, I doubt we'll run into the issue though